pkg/triggerconfig/merge/merge.go

```diff
@@ -0,0 +1,46 @@
+package merge
+
+import (
+	"github.com/jenkins-x/lighthouse/pkg/config"
+	"github.com/jenkins-x/lighthouse/pkg/config/job"
+	"github.com/jenkins-x/lighthouse/pkg/plugins"
+	"github.com/jenkins-x/lighthouse/pkg/triggerconfig"
+)
+
+// ConfigMerge merges the repository configuration into the global configuration
+func ConfigMerge(cfg *config.Config, pluginsCfg *plugins.Configuration, repoConfig *triggerconfig.Config, repoOwner string, repoName string) error {
+	repoKey := repoOwner + "/" + repoName
+	if len(repoConfig.Spec.Presubmits) > 0 {
+		if cfg.Presubmits == nil {
+			cfg.Presubmits = map[string][]job.Presubmit{}
+		}
+
+		var ps []job.Presubmit
+		for _, p := range repoConfig.Spec.Presubmits {
+			ps = append(ps, ToPresubmit(p))
```

Comment:
There are probably other constraints to check, like no duplicate job name, etc... recent refactoring should expose a set of methods that can help with individual jobs (`SetDefaults`, `SetRegexes`, `Validate`, etc...) but they also need to be validated as a whole.

Reply:
we call the `Validate()` methods on lighthouse + prow configs